hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/RecordStore.java

```diff
@@ -63,7 +64,7 @@
      */
     Object put(Data dataKey, Object dataValue, long ttl);
 
-    Object putIfAbsent(Data dataKey, Object value, long ttl);
+    Object putIfAbsent(Data dataKey, Object value, long ttl, Address callerAddress);
```

Comment:
I somewhat dislike that we now provide the `callerAddress` to the record store and that it is now responsible for publishing events but I don't see any way around it. WDYT?

Reply:
We can also do this at operations' level but this seemed the simplest way since both EE and OS operations are using this method.